packages/playwright-core/src/server/firefox/ffPage.ts

```diff
@@ -394,7 +395,8 @@ export class FFPage implements PageDelegate {
   }
 
   async evaluateOnNewDocument(source: string): Promise<void> {
-    await this._session.send('Page.addScriptToEvaluateOnNewDocument', { script: source });
+    this._initScripts.push(source);
+    await this._session.send('Page.setInitScripts', { scripts: this._initScripts.map(script => ({ script })) });
```

Comment:
Shouldn't this include the UTILITY_WORLD_NAME empty script as well?

Reply:
I'm not sure!